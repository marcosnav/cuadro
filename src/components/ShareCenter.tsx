import React, { FC } from 'react';
import { getShareFunc, SharePlatform } from './../utils';
import PrettyLink from './PrettyLink';
import * as S from './styled';

const ShareCenter: FC = () => {
  return (
    <S.Section>
      <S.CapsTitle>
        Share The Joy
      </S.CapsTitle>
      <S.ShareSection>
        <PrettyLink onClick={getShareFunc(SharePlatform.FACEBOOK)}>
          Facebook
        </PrettyLink>
        <PrettyLink onClick={getShareFunc(SharePlatform.TWITTER)}>
          Twitter
        </PrettyLink>
      </S.ShareSection>
    </S.Section>
  );
};

export default ShareCenter;
