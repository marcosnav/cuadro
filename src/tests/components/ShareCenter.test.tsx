import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import ShareCenter from './../../components/ShareCenter';

describe('<ShareCenter/>', () => {
  it('renders properly', () => {
    const component = mount(<ShareCenter/>);
    expect(component).toMatchSnapshot();
  });
});
