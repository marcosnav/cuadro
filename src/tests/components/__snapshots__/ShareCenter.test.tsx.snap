// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareCenter/> renders properly 1`] = `
.c3 {
  font-weight: bold;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
}

.c3:hover {
  opacity: 0.4;
}

.c1 {
  margin: 0;
  text-align: center;
  text-transform: uppercase;
}

.c0 {
  margin-top: 42px;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.83em;
  margin-top: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 300px;
}

.c2 > * {
  margin: 0 10px;
}

<Section>
  <section
    className="c0"
  >
    <CapsTitle>
      <h4
        className="c1"
      >
        Share The Joy
      </h4>
    </CapsTitle>
    <ShareSection>
      <div
        className="c2"
      >
        <Link
          onClick={[Function]}
          target="_blank"
        >
          <a
            className="c3"
            onClick={[Function]}
            target="_blank"
          >
            Facebook
          </a>
        </Link>
        <Link
          onClick={[Function]}
          target="_blank"
        >
          <a
            className="c3"
            onClick={[Function]}
            target="_blank"
          >
            Twitter
          </a>
        </Link>
      </div>
    </ShareSection>
  </section>
</Section>
`;
